packages/build-tools/src/common/prebuild.ts

```diff
@@ -32,9 +32,13 @@ export async function prebuildAsync<TJob extends BuildJob>(
     packageManager: ctx.packageManager,
   });
   const installDependenciesSpawnPromise = (
-    await installDependenciesAsync(ctx, {
+    await installDependenciesAsync({
+      packageManager: ctx.packageManager,
+      env: ctx.env,
       logger,
       cwd: resolvePackagerDir(ctx),
+      // prebuild sometimes modifies package.json, so we don't want to use frozen lockfile
+      useFrozenLockfile: false,
```

Comment:
i believe it only modifies package.json if using a custom template, we tried to move away from modifying dependencies in prebuild in part to fix this problem of not being able to use frozen lockfiles. perhaps @byCedric has the most current context on this